### PR TITLE
add entities

### PR DESCRIPTION
**Pull request by daria-sieroshtan**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,7 @@
+deploy_prod:
+	rm -rf var/cache/*
+	php bin/console doctrine:migrations:migrate -n
+
```
Comment (arodiss) [inline Makefile]: @daria-sieroshtan 
the content does not match the label

File: src/Entity/ExerciseTag.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+     * @ORM\ManyToMany(targetEntity=Exercise::class, mappedBy="tags")
+     */
+    private $exercises;
+
+    public function __construct()
+    {
+        $this->exercises = new ArrayCollection();
+    }
+
+    public function getId(): ?int
```
Comment (arodiss) [inline src/Entity/ExerciseTag.php:47]: @daria-sieroshtan 
Can this method return null?

**Review by arodiss — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -8,3 +8,9 @@ test:
 
 check_code_style:
 	php vendor/bin/phpcs --standard=psr12 src/ -n
+
+clear_cache:
+	rm -rf var/cache/*
+
+migrate_doctrine_migrations:
```
Comment (arodiss) [inline Makefile]: it's up to you but I would go with just "migrate"

File: src/Controller/BaseController.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+     * @return void
+     */
+    public function addNoticeFlash($message)
+    {
+        $this->addFlash('notice', $message);
+    }
+    public function addSuccessFlash($message)
+    {
+        $this->addFlash('success', $message);
+    }
```
Comment (arodiss) [inline src/Controller/BaseController.php:35]: NL between methods
Does your cs check work on this?

File: src/Controller/BaseController.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        $entityManager = $this->getDoctrine()->getManager();
+        $entityManager->remove($entity);
+        $entityManager->flush();
+    }
+
+    /**
+     * @param Request $request
+     * @return string|string[]|null
+     */
+    public function getRedirectToPrevious(Request $request)
```
Comment (arodiss) [inline src/Controller/BaseController.php]: @daria-sieroshtan 
this method mentions redirect but actualy has nothing about it.
I assume redirect is how you use this method.
If this is the case - think about the letter "D" in SOLID :)

File: src/Controller/ExerciseController.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+ */
+class ExerciseController extends BaseController
+{
+    /**
+     * @Route("/", name="exercise_index", methods={"GET"})
+     */
+    public function index(ExerciseRepository $exerciseRepository): Response
+    {
+        return $this->render('exercise/index.html.twig', [
+            'exercises' => $exerciseRepository->findAll(),
```
Comment (arodiss) [inline src/Controller/ExerciseController.php:23]: @daria-sieroshtan 
I know this is not final but you may want to add todo for pagination

File: src/Controller/ExerciseController.php [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    public function new(Request $request): Response
+    {
+        $exercise = new Exercise();
+        $form = $this->createForm(ExerciseType::class, $exercise);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            if ($form->get('intervals')->getData() == null) {
+                $exercise->setIntervals(1);
+            }
```
Comment (arodiss) [inline src/Controller/ExerciseController.php]: @daria-sieroshtan 
This does belong to the controller.
You can use form event, data transformer, or doctrine event to handle this properly.
You should also have DB constraint if you rely on this thing not being null

File: src/Controller/ExerciseController.php [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->persist($exercise);
+            $entityManager->flush();
```
Comment (arodiss) [inline src/Controller/ExerciseController.php]: @daria-sieroshtan 
you created a helper method for this

File: src/Controller/ExerciseController.php [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{id}", name="exercise_show", methods={"GET"})
+     */
+    public function show(Exercise $exercise): Response
+    {
+        return $this->render('exercise/show.html.twig', [
```
Comment (arodiss) [inline src/Controller/ExerciseController.php:55]: @daria-sieroshtan 
what you are doing is fine but I would like to remind you that you can use `@Template` annotation for clarity

File: src/Entity/Exercise.php [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+     * @ORM\Column(type="integer")
+     * @Assert\LessThanOrEqual(6)
+     * @Assert\GreaterThanOrEqual(1)
+     */
+    private $intervals;
+
+    /**
+     * @ORM\Column(type="boolean")
+     */
+    private $isPrivate;
```
Comment (arodiss) [inline src/Entity/Exercise.php]: @daria-sieroshtan 
in this app it does not matter but generally, you should always set a default value for booleans.
Nullable boolean = 2 bits, non-nullable-boolean = 1 bit
This makes DB queries significantly faster

File: templates/base.html.twig [hunk 11]
```diff
@@ -2,11 +2,96 @@
...
-        {% block body %}{% endblock %}
-        {% block javascripts %}{% endblock %}
+    {% block body %}
+        <div class="container-fluid">
+            {% block navbar %}
+
+                <nav class="navbar navbar-light bg-white fixed-top border-bottom">
+                    <a class="navbar-brand text-dark" href="{{ path('home') }}">
+                        <img src="/img/exercise-icon-16327.png" width="30" height="30" alt="" class="d-inline-block align-top">
+                        &nbspExerclist
```
Comment (arodiss) [inline templates/base.html.twig]: @daria-sieroshtan 
`&nbsp;` (notice semicolon)
There are more below

**Review by arodiss — CHANGES_REQUESTED** [review]

File: src/Controller/BaseController.php (hunk 4 above)
Comment (daria-sieroshtan) [reply src/Controller/BaseController.php:35]: cs doesn't complain about it, but I made the changes nevertheless.

File: src/Controller/ExerciseController.php (hunk 6 above)
Comment (daria-sieroshtan) [reply src/Controller/ExerciseController.php:23]: Already in todo.

File: src/Controller/ExerciseController.php (hunk 9 above)
Comment (daria-sieroshtan) [reply src/Controller/ExerciseController.php:55]: I do remember.
This part is auto-generated so I left it as is.

File: Makefile [hunk 12]
```diff
@@ -8,3 +8,19 @@ test:
...
+
+migrate:
+	php bin/console doctrine:migrations:migrate -n
+
+#docker part is wip
+build_docker:
+	docker build . -t exerclist
+
+run_docker:
+	docker run -p 8000:8000 exerclist
```
Comment (arodiss) [inline Makefile:26]: @daria-sieroshtan 
there is a problem with this.

File: src/Controller/HomeController.php [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+     * @Route("/",
+     *      name="home",
+     *     )
+     * @Template("home.html.twig")
+     *
+     */
+    public function home()
+    {
+        return [];
+    }
```
Comment (arodiss) [inline src/Controller/HomeController.php:28]: @daria-sieroshtan 
are you sure you want to have an empty home screen? Few options:
- Put core functionality there, like "run a workout"
- Show some stats
It's not mandatory to do it today

File: src/Controller/WorkoutController.php [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    /**
+     * @Route("/new", name="workout_new", methods={"GET","POST"})
+     */
+    public function new(Request $request): Response
+    {
+        $workout = new Workout();
+//        todo: proper workout creation
+        $workout->setDuration(1);
```
Comment (arodiss) [inline src/Controller/WorkoutController.php]: @daria-sieroshtan 
errrr... what?!

**Review by arodiss — CHANGES_REQUESTED** [review]

**Review by arodiss — APPROVED** [review]

**Pull request merged**